experimental/devices/rainbowhat/rainbowhat.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package rainbowhat
+
+import (
+	"periph.io/x/periph/conn/spi/spireg"
+	"periph.io/x/periph/devices/apa102"
+	"periph.io/x/periph/devices/bmxx80"
+	"periph.io/x/periph/experimental/devices/ht16k33"
+	"periph.io/x/periph/host/rpi"
+
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c/i2creg"
+)
+
+const (
+	numPixels  = 7
+	bmp280Addr = 0x77
+)
+
+// Dev represents a Rainbow HAT  (https://shop.pimoroni.com/products/rainbow-hat-for-android-things)
+type Dev struct {
+	ledstrip *apa102.Dev
```

Comment:
That's really nice composition work!

Personally, I'd make these public; this way you don't need the GetFoo() methods and it's easier to use IMO.

Reply:
Yes there's a risk but this is fine. This is not java, we do trust the developer.

Reply:
hahaha good point